### PR TITLE
[RHCLOUD-23167] feature: return the "meta" information for the behavior group collection

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
This information was missing from the collection's response, and QE reported it, so this is why we are including it in the responses from now on.

## Links

[[RHCLOUD-23167]](https://issues.redhat.com/browse/RHCLOUD-21880)

### CODE REVIEW

**Comment by josejulio** [comment]
> @jmartine Do you mean that we would publish `/api/notifications/v1.1`?

Yes, I'm unsure how that works or is setup - but we have this PR and other blocked because we require some frontend changes. Having a different version would help in that regard.

File: backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java [hunk 1]
```diff
@@ -79,12 +79,47 @@ public class NotificationResource {
...
+        @GET
+        @Path("/eventTypes/{eventTypeId}/behaviorGroups")
+        @Produces(APPLICATION_JSON)
+        @Operation(summary = "Retrieve the behavior groups linked to an event type.")
+        @RolesAllowed(ConsoleIdentityProvider.RBAC_READ_NOTIFICATIONS)
+        public List<BehaviorGroup> getLinkedBehaviorGroups(
+                @Context SecurityContext sec,
+                @PathParam("eventTypeId") UUID eventTypeId,
+                @BeanParam @Valid Query query,
+                @Context UriInfo uriInfo
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java:91]:
```suggestion
                @BeanParam @Valid Query query
```

File: backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java [hunk 2]
```diff
@@ -79,12 +79,47 @@ public class NotificationResource {
...
+        ) {
+            String orgId = getOrgId(sec);
+
+            final List<BehaviorGroup> behaviorGroups = this.behaviorGroupRepository.findBehaviorGroupsByEventTypeId(orgId, eventTypeId, query);
+            final long behaviorGroupCount = this.behaviorGroupRepository.countByEventTypeId(orgId, eventTypeId);
+
+            return new Page<>(
+                    behaviorGroups,
+                    PageLinksBuilder.build(uriInfo.getPath(), behaviorGroupCount, query.getLimit().getLimit(), query.getLimit().getOffset()),
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java:119]:
```suggestion
                    PageLinksBuilder.build(uriInfo.getPath(), behaviorGroupCount, query),
```

File: backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java [hunk 3]
```diff
@@ -669,6 +675,31 @@ private void checkEventTypeBehaviorGroups(Header identityHeader, String eventTyp
...
+                .header(identityHeader)
+                .pathParam("eventTypeId", eventTypeId)
+                .when()
+                .get("/notifications/eventTypes/{eventTypeId}/behaviorGroups")
+                .then()
+                .statusCode(200)
+                .contentType(JSON)
+                .extract().body().asString();
+
+        // The response body contains the meta information about the history collection. We need to access the
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java:690]:
```suggestion
        // The response body contains the meta information about the behavior groups collection. We need to access the
```

File: backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java [hunk 4]
```diff
@@ -669,6 +675,31 @@ private void checkEventTypeBehaviorGroups(Header identityHeader, String eventTyp
...
+        // "data" array to access the elements.
+        final JsonObject responseBodyJson = new JsonObject(responseBody);
+        final JsonArray jsonBehaviorGroups = responseBodyJson.getJsonArray("data");
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java:693]: This test could check the other fields from `responseBodyJson` (`meta` and `links`).

File: backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java (hunk 1 above)
Comment (MikelAlejoBR) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java]: Done!

File: backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java (hunk 2 above)
Comment (MikelAlejoBR) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java]: Done!

File: backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java (hunk 3 above)
Comment (MikelAlejoBR) [reply backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java]: Done!

File: backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java (hunk 4 above)
Comment (MikelAlejoBR) [reply backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java:694]: Done!

File: backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java [hunk 5]
```diff
@@ -669,6 +675,44 @@ private void checkEventTypeBehaviorGroups(Header identityHeader, String eventTyp
...
+        assertNotNull(links, "the \"links\" element is missing from the paged response");
+
+        // Check the "first" link.
+        final String expectedFirstValue = String.format("/api/notifications/v2.0/notifications/eventTypes/%s/behaviorGroups?limit=20&offset=0", eventTypeId);
+        assertEquals(expectedFirstValue, links.getString("first"), "the \"first\" link element contains an unexpected value");
+
+        // Check the "last" link.
+        final String expectedLastValue = String.format("/api/notifications/v2.0/notifications/eventTypes/%s/behaviorGroups?limit=20&offset=0", eventTypeId);
+        assertEquals(expectedLastValue, links.getString("last"), "the \"last\" link element contains an unexpected value");
+    }
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java:714]:
```suggestion

        final JsonObject meta = responseBodyJson.getJsonObject("meta");
        assertEquals(expectedBehaviorGroupIds.length, meta.getNumber("count"));
    }
```

Comment (MikelAlejoBR) [reply backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java:718]: OMG! I forgot about this one :sweat_smile:

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**